src/databases/WavefrontOBJ/avtWavefrontOBJWriter.C

```diff
@@ -173,3 +211,56 @@ avtWavefrontOBJWriter::GetCombineMode(const std::string &) const
 {
     return CombineAll;
 }
+
+//****************************************************************************
+// Method:  avtWavefrontOBJWriter::GetColorTable()
+//
+// Purpose:
+//   Create color table
+//
+//
+// Programmer:  Justin Privitera
+// Creation:    11/03/23
+//
+// Modifications:
+//
+//****************************************************************************
+
+vtkImageData *
+avtWavefrontOBJWriter::GetColorTable()
+{
+    const ColorTableAttributes *colorTables = avtColorTables::Instance()->GetColorTables();
+    const ColorControlPointList *table = colorTables->GetColorControlPoints(colorTable);
+    if (table)
+    {
+        unsigned char rgb[256*3];
+        int ncolors = 256;
+        table->GetColors(rgb, ncolors);
+
+        vtkImageData *imageData = vtkImageData::New();
+
+        // TODO change image extent dynamically based on num_ctrl_pts
+        // x: [0, ncolors - 1], y: [0, 0], z: [0, 0]
+        imageData->SetExtent(0, ncolors - 1, 0, 0, 0, 0);
+        imageData->SetSpacing(1., 1., 1.);
+        imageData->SetOrigin(0., 0., 0.);
+        imageData->AllocateScalars(VTK_UNSIGNED_CHAR, 3);
+        unsigned char *pixels = (unsigned char *)imageData->GetScalarPointer(0, 0, 0);
+        unsigned char *ipixel = pixels;
```

Comment:
I am a little unclear what this is doing...it appears to be creating a teeny, tiny `vtkImageData` thingy with each pixel one of the needed colors from a color table. I am guessing this has something to do with the way OBJ files or VTK's OBJ writer does its work. I think it could help to have a description about why this is necessary because on the face of it, it doesn't make sense.

Reply:
Yes, I am making a 1D strip of 256 pixels that approximates a color table. I have to stuff this information into a `vtkImageData`. This is needed to appease the `vtkPNGWriter`, which I am using to get out a texture. I can put a note in to clarify.